stl/inc/vector

```diff
@@ -982,22 +1166,39 @@ public:
         pointer& _Myfirst = _My_data._Myfirst;
         pointer& _Mylast  = _My_data._Mylast;
 
+        constexpr bool _Nothrow_construct =
+            conjunction_v<is_nothrow_copy_constructible<_Ty>, _Uses_default_construct<_Alloc, _Ty*, const _Ty&>>;
+
         _My_data._Orphan_all();
         const auto _Oldcapacity = static_cast<size_type>(_My_data._Myend - _Myfirst);
         if (_Newsize > _Oldcapacity) { // reallocate
             _Clear_and_reserve_geometric(_Newsize);
-            _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
+            if constexpr (_Nothrow_construct) {
+                _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
+                _ASAN_VECTOR_CREATE;
+            } else {
+                _ASAN_VECTOR_GUARD_CREATE;
+                _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
+            }

```

Comment:
```suggestion
            _TRY_BEGIN
            _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
            _CATCH_ALL
            _ASAN_VECTOR_CREATE;
            _CATCH_END
            _ASAN_VECTOR_CREATE;
```

Reply:
Why bother with `try`/`catch` like this instead of having an RAII guard that `_ASAN_VECTOR_CREATE`s in its destructor? At the very least, we wouldn't have to remember to `_RETHROW` in the catch block =)